File: typescript/src/main/vsdk/toolkit/material/BsdfComponentFlag.ts
import { XxdfComponentFlagInfo } from "./XxdfComponentFlag";

export class BsdfComponentFlag {
  private constructor() {
  }

  public static bsdfIndexToComp(index: number): number {
    return 1 << index;
  }

  public static getBrdfFlags(bsflags: number): number {
    return bsflags & XxdfComponentFlagInfo.ALL_COMPONENTS;
  }

  public static getBtdfFlags(bsflags: number): number {
    return (bsflags >> XxdfComponentFlagInfo.XXDF_COMPONENTS) & XxdfComponentFlagInfo.ALL_COMPONENTS;
  }
}

File: java/base/src/main/java/vsdk/toolkit/material/BsdfComponentFlag.java
package vsdk.toolkit.material;

public final class BsdfComponentFlag {
    private BsdfComponentFlag() {
    }

    public static int bsdfIndexToComp(int index) {
        return 1 << index;
    }

    public static int getBrdfFlags(int bsflags) {
        return bsflags & XxdfComponentFlagInfo.ALL_COMPONENTS;
    }

    public static int getBtdfFlags(int bsflags) {
        return (bsflags >> XxdfComponentFlagInfo.XXDF_COMPONENTS) & XxdfComponentFlagInfo.ALL_COMPONENTS;
    }
}
